test/evergreen/compatibility_test_for_releases.sh

```diff
@@ -157,37 +189,127 @@ create_configs_for_newer_release_branches()
 #############################################################
 run_format()
 {
-        branch_name=$1
+    branch_name=$1
+    echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+    echo "Running format in branch: \"$branch_name\""
+    echo "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
+
+    cd "$branch_name/test/format"
+    flags="-1q $(bflag $branch_name)"
+
+    config_file=""
+
+    # Compatibility test for newer releases will have CONFIG file for each release
+    # branches for the upgrade/downgrade testing.
+    #
+    # Compatibility test for older and standalone releases will have the default config.
+    if [ "${wt_standalone}" = true ] || [ $older = true ]; then
+        config_file="-c CONFIG_default"
+    else
+        config_file="-c CONFIG_${branch_name}"
+    fi
+
+    for am in $2; do
+        dir="RUNDIR.$am"
+        echo "./t running $am access method..."
+        ./t $flags ${config_file} -h $dir "file_type=$am"
+
+        # Remove the version string from the base configuration file. (MongoDB does not create
+        # a base configuration file, but format does, so we need to remove its version string
+        # to allow backward compatibility testing.)
+        (echo '/^version=/d'
+            echo w) | ed -s $dir/WiredTiger.basecfg > /dev/null
+    done
+    cd -
+}
+
+#############################################################
+# is_test_checkpoint_recovery_supported:
+#       arg1: branch name
+#############################################################
+is_test_checkpoint_recovery_supported()
+{
+    branch_name=$1
+
+    #
+    # The test_checkpoint recovery change WT-7958 are included in mongo
+    # 4.4.9 and 5.0.3 (and newer) versions.
+    #
+    # There's no much value to run test_checkpoint for earlier versions.
+    #
+    # Exclude mongodb-4.4.10+ and mongodb-5.0.10+ versions.
+    #
+    # Exclude mongodb-4.4 and mongodb-5.0 and they represent latest code
+    # of the corresponding release branches, which have WT-7958 included.
+    #
+    if ( ([[ $branch_name == mongodb-* ]] &&
+          [[ $branch_name < "mongodb-4.4.9" ]]) ||
+         ([[ $branch_name == mongodb-5.0.[0-9] ]] &&
```

Comment:
Does it have to be `=~` instead of `==` to work with regex?

Reply:
My testing shows the above `==` works. I'm not totally sure if `=~` is a requirement for regex match in shell though.